tests/UI/pages/BO/advancedParameters/APIAccess/add.ts

```diff
@@ -83,12 +89,84 @@ class AddNewAPIAccess extends BOBasePage {
     await this.setValue(page, this.clientIdInput, apiAccessData.clientId);
     await this.setValue(page, this.descriptionInput, apiAccessData.description);
     await this.setValue(page, this.tokenLifetimeInput, apiAccessData.tokenLifetime);
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const scope of apiAccessData.scopes) {
+      await this.setAPIScopeChecked(page, scope, true);
+    }
+
     // Save
     await this.clickAndWaitForURL(page, this.saveButton);
 
     return this.getAlertSuccessBlockParagraphContent(page);
   }
 
+  /**
+   * Returns the list of scopes from a group
+   * @param page {Page} Browser tab
+   * @param group {string} Scopes Group
+   */
+  async getApiScopes(page: Page, group: string): Promise<string[]> {
+    const scopes: string[] = [];
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const scope of await page.locator(`#api_access_scopes_${group}_accordion div.switch-scope`).all()) {
+      const dataScope: string|null = await scope.getAttribute('data-scope');
+
+      if (dataScope) {
+        scopes.push(dataScope);
+      }
+    }
+
+    return scopes;
```

Comment:
1. Maybe declare this selector like this `this.accessScopBlock = (group) => `#api_access_scopes_${group}_accordion div.switch-scope`;

2. Why not use `evaluateAll` like this : 
```js
scops = await page.locator(this.accessScopBlock)
  .evaluateAll(
    (all) => all
      .map(
        (el) => el.getAttribute('data-scope')
      )
      .filter(
        (attr) => attr != undefined
      )
    );
```

Reply:
I check if it's ok.

Reply:
Fixed